src/home/pam_systemd_home.c

```diff
@@ -826,6 +826,12 @@ _public_ PAM_EXTERN int pam_sm_open_session(
         if (r != PAM_SUCCESS)
                 return r;
 
+        /* Explicitly get saved PamBusData here. Otherwise, this function may succeed without setting 'd'
+         * even if there is an opened sd-bus connection, and it will be leaked. See issue #31375. */
+        r = pam_get_bus_data(handle, "pam-systemd-home", &d);
```

Comment:
Hmm, with this, it's no longer necessary for `acquire_home` or `acquire_user_record` to return `PamBusData`?

Reply:
It's necessary, as at this time there may be no bus connection, and `d` is NULL in that case. And a new bus connection may be opened in acquire_home() later below.

Reply:
Indeed... But it's easier to follow if there's only one source of truth. So maybe just introduce a cleanup func that does `pam_get_bus_data` + `pam_bus_data_disconnectp`?

Reply:
> Such hack (pam_get_bus_data + pam_bus_data_disconnectp) is only necessary when this function succeeds without calling pam_acquire_bus_connection(). 

I generally agree, but `pam_get_bus_data` is now unconditionally called, so it kinda feels weird to me.

But doesn't matter too much after all.